hydra/test_utils/launcher_common_tests.py

```diff
@@ -431,6 +431,26 @@ def get_test_app_working_dir(self) -> Optional[Path]:
         """
         return None
 
+    def get_test_scratch_dir(self, tmpdir: Path) -> Path:
+        """
+        By default test applications will use tmpdir provided by the pytest.
+        This can be customized by applications.
+        """
+        return tmpdir
+
+    def generate_custom_cmd(self) -> Callable[..., List[str]]:
+        """
+        By default this does nothing, but it allows custom execution commands.
+        Useful if the tests are not kicked off by python
+        """
+
+        def fun(
+            cmd: List[str], tmpdir: Path, modified_tmpdir: Path, filename: str
```

Comment:
1. Please document the parameters, they are not obvious
2. Why do you need tempdir and modified modified_tmpdir here?

Reply:
updated in the new commit